src/internal.c

```diff
@@ -21931,6 +22095,16 @@ static int DoProcessReplyEx(WOLFSSL* ssl, int allowSocketErr)
                 used = ssl->buffers.inputBuffer.length -
                        ssl->buffers.inputBuffer.idx;
                 if (used < readSz) {
+                    if (used > 0 && !DtlsRecordsCanSpanDatagrams(ssl)) {
+                        WOLFSSL_MSG("DTLS: Partial record in buffer, dropping");
+                        ssl->options.processReply = doProcessInit;
+                        ssl->buffers.inputBuffer.length = 0;
+                        ssl->buffers.inputBuffer.idx = 0;
+#ifdef WOLFSSL_DTLS_DROP_STATS
+                        ssl->replayDropCount++;
+#endif /* WOLFSSL_DTLS_DROP_STATS */
```

Comment:
This exact pattern is repeated multiple times in multiple locations. It would be nice to refactor it into either a `void func` or a macro.

Reply:
done